portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,40 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList; // import the ArrayList class
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Andrew!");
+    ArrayList<String> flavors = new ArrayList<String>();
+    flavors.add("Cookies and Cream");  
+    flavors.add("Vanilla");  
+    flavors.add("Mint Chocolate Chip");  
+
+    String json = convertToJson(flavors);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+    /**
+   * Converts Array List instance into a JSON string using String concatentation.
+   */
+  private String convertToJson(ArrayList<String> flavors) {
```

Comment:
I see you added the GSON dependency but you are not using it to convert the ArrayList to JSON. From the walkthrough it seems like you can do either but it might be useful to call it so you have practice calling some external libraries